x-pack/plugins/security_solution/server/lib/detection_engine/signals/rule_status_saved_objects_client.ts

```diff
@@ -35,6 +42,50 @@ export const ruleStatusSavedObjectsClientFactory = (
       ...options,
       type: ruleStatusSavedObjectType,
     }),
+  findBulk: async (ids, statusesPerId) => {
+    if (ids.length === 0) {
+      return {};
+    }
+    const filter = buildChunkedOrFilter(`${ruleStatusSavedObjectType}.attributes.alertId`, ids);
+    const order: 'desc' = 'desc';
+    const aggs = {
+      alertIds: {
+        terms: {
+          field: `${ruleStatusSavedObjectType}.attributes.alertId`,
+          size: ids.length,
+        },
+        aggs: {
+          most_recent_statuses: {
+            top_hits: {
+              sort: [
+                {
+                  [`${ruleStatusSavedObjectType}.statusDate`]: {
+                    order,
+                  },
+                },
+              ],
+              size: statusesPerId,
+            },
+          },
+        },
+      },
+    };
+    const results = await savedObjectsClient.find({
+      filter,
+      aggs,
+      type: ruleStatusSavedObjectType,
+      perPage: 0,
+    });
+    const buckets = get(results, 'aggregations.alertIds.buckets');
+    return buckets.reduce((acc: Record<string, unknown>, bucket: unknown) => {
+      const key = get(bucket, 'key');
+      const hits = get(bucket, 'most_recent_statuses.hits.hits');
```

Comment:
nit: I would try to cast here if possible maybe? `const hits: someType` even if that type was something loose such as `Record<string, unknown>`. I didn't play with it, but less `any` is good where/when we can use `unknown` or some other type.